### PR TITLE
Name mismatch screen and user input validation

### PR DESCRIPTION
**Pull request by eedwards0**
### IDVA3-626 and IDVA3-1731 - Name mismatch reason screen and user input validation

1. Add web page
2. Add user input validation when no reason selected
3. Add unit and integration tests
4. To aid with testing, a temporary update to PersonalCodeHandler diverts to the name mismatch screen with UVID "111-222-333"
5. Update `api-sdk-node` to 2.0.217 for latest NameMismatchReasonEnum

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/views/router_views/name_mismatch/name_mismatch.njk [hunk 1]
```diff
@@ -0,0 +1,112 @@
+{% extends "layouts/default.njk" %}
+
+{% from "govuk/components/button/macro.njk" import govukButton %}
+{% from "govuk/components/details/macro.njk" import govukDetails %}
```
Comment (PhilipGlover) [inline src/views/router_views/name_mismatch/name_mismatch.njk:4]: `govukDetails` isn't used in this template

File: src/constants.ts [hunk 2]
```diff
@@ -38,6 +38,7 @@ export const Urls = {
     INDIVIDUAL_PSC_LIST: "/individual/psc-list",
     NEW_SUBMISSION: "/new-submission",
     PERSONAL_CODE: `${urlWithTransactionIdAndSubmissionId}/individual/personal-code`,
+    NAME_MISMATCH: `${urlWithTransactionIdAndSubmissionId}/individual/name-mismatch`,
```
Comment (PhilipGlover) [inline src/constants.ts:41]: I wonder if we should be following the prototype here and use the path `/individual/psc-why-this-name` ?

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by PhilipGlover — COMMENTED** [review]
- have run journey successfully, using the personal code '111-2222-3333'
- I see the Verification doc being updated in db, for all the different Mismatch types eg
```
        "verification_details" : {
            "uvid" : "111-2222-3333",
            "name_mismatch_reason" : "PREFER_NOT_TO_SAY",
            "statements" : []
        }
```
- validation on the screen works if no option selected
- tests pass and look good
- my only hesitation / query to others is the addition of the '111-2222-3333' code in the `personalCodeHandler`.  I understand we can take it out in the future Validation story.  But as far as testing is concerned we could just say to use `/individual/psc-why-this-name` after the personal code screen.  That's what was done for the Stop screens.  I don't have any strong opinion either way,  as long as we're all happy.

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Pull request merged**